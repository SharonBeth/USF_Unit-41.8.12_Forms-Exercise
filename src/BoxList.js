import './App.css';
import React, { useState } from "react";
import { v4 as uuid } from 'uuid';
import NewBoxForm from './NewBoxForm'
import Box from './Box';


function BoxList() {
    const INITIAL_STATE = [
        // { id: uuid(), width: 34, height: 72, backgroundColor: 'orange' },
        // { id: uuid(), width: 40, height: 80, backgroundColor: 'green' }
    ]
    const [boxes, setBoxes] = useState(INITIAL_STATE);
    const addBox = (newBox) => {
        setBoxes(boxes => [...boxes, { ...newBox, id: uuid() }])
    }

    const remove = id => {
        setBoxes(boxes => boxes.filter(box => box.id !== id));
    };

    return (
        <div className="BoxList">
            <h3>Color Box Maker</h3>
            <NewBoxForm addBox={addBox} />
            <div>
                {boxes.map(({ id, width, height, backgroundColor }) => <Box id={id} width={width} height={height} backgroundColor={backgroundColor} key={id} handleRemove={remove} />)}
            </div>
        </div>
    );
}

export default BoxList;
