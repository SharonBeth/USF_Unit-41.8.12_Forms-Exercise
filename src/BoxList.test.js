import React from "react";
import { render, fireEvent } from "@testing-library/react";
import BoxList from "./BoxList";

it("renders without crashing-Smoke Test", function () {
    render(<BoxList />)
});

it("passes the snapshot", function () {
    const { asFragment } = render(<BoxList />)
    expect(asFragment()).toMatchSnapshot();
})

it("should add width", function () {
    const { queryByText, getByLabelText } = render(<BoxList />);
    const input = getByLabelText("Width");
    const btn = queryByText("Add Item")
    expect(queryByText('Width: 9')).not.toBeInTheDocument();
    fireEvent.change(input, { target: { value: '9' } });
    fireEvent.click(btn);
    expect(queryByText('Width: 9')).toBeInTheDocument();
})

it("should add backgroundColor and remove it", function () {
    const { queryByText, getByLabelText } = render(<BoxList />);
    const input = getByLabelText("Background Color");
    const btn = queryByText("Add Item")
    expect(queryByText('Background Color: Green')).not.toBeInTheDocument();
    fireEvent.change(input, { target: { value: 'Green' } });
    fireEvent.click(btn, { target: { value: "green" } });
    expect(queryByText('Background Color: Green')).toBeInTheDocument();
    const btn2 = queryByText("Remove the Box!")
    fireEvent.click(btn2)
    expect(queryByText('Background Color: Green')).not.toBeInTheDocument();


})