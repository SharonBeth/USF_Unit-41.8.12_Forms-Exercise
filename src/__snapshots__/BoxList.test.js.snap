// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`passes the snapshot 1`] = `
<DocumentFragment>
  <div
    class="BoxList"
  >
    <h3>
      Color Box Maker
    </h3>
    <form>
      <label
        for="width"
      >
        Width
      </label>
      <input
        id="width"
        max="400"
        min="1"
        name="width"
        placeholder="Width"
        type="integer"
        value=""
      />
      <label
        for="height"
      >
        Height
      </label>
      <input
        id="height"
        max="400"
        min="1"
        name="height"
        type="integer"
        value=""
      />
      <label
        for="backgroundColor"
      >
        Background Color
      </label>
      <input
        id="backgroundColor"
        name="backgroundColor"
        type="text"
        value=""
      />
      <button>
        Add Item
      </button>
    </form>
    <div />
  </div>
</DocumentFragment>
`;
